### DISCUSSION TITLE
New icon font size for Nextion

### DISCUSSION THREAD

**Discussion by Floppe** [Blackymas/NSPanel_HA_Blueprint#838] [General]
Hi!

Some help with Nextion editor. Could you give me hints on how to add a new icon font with 64 size so it matches same index set as the other fonts?

Thanks

**Comment by edwardtfn** [comment]
Are you still looking for an answer to this? If so, could you please explain better what are you trying to achieve?

**Comment by Floppe** [comment]
My first version of my own energy page was with icons from the font so the end user could chose own icons, but they were so small. I later changed to static icons by using background image masking. So no need anymore right now.

Here a screenshot

![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/238729/022ae798-1d2e-4543-b281-b6d133c8dca3)

Current energy usage and buy, sell and spot price info. You've done so much changes for version 4 so I probably need to start from scratch again.

**Reply by edwardtfn** [reply]
That looks awesome?
Anything I could help you?

**Reply by edwardtfn** [reply]
I can generate a font with bigger icons, but those fonts increases the TFT by a lot. I'm already thinking about changing one of the icon's fonts to size 64, which might attend to your need.

**Reply by edwardtfn** [reply]
By the way, with v3.4, the icon font was with sizes 10, 24 and 32. With v4.0 I'm adding size 48, but still considering change that to size 64. I cannot have both as the file starts getting too big.
But maybe size 48 is enough for your case.

**Reply by edwardtfn** [reply]
Also, back to your original question, you can find the font size 64 here: https://github.com/joBr99/Generate-HASP-Fonts/tree/master/output
We are using NotoSans for icons.

On Nextion Editor, select the tab with the fonts list on that same area where the pictures are:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/6f17a4b4-d56c-4148-b9b2-020d878ecc8a)

Then find in the list the font you want to replace, right click and then "Replace":
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/0ebdf928-72a3-479e-ba72-e98beb5e8730)

Then select the new font file you want to use. All the components using the older font will start using the new one, in all pages.

You can also just add a new font, so you use that only on this page you are creating. Just be careful with the final tft file size.

**Reply by edwardtfn** [reply]
> You've done so much changes for version 4 so I probably need to start from scratch again.

You don't need to start from scratch. You can always right click at the page name and export to a file, then import later to v4.0.
I would love to get more details about your solution so we can merge into `dev` and later to `main`, so it would be available to everyone. Only if you are OK with that, of course.

**Comment by edwardtfn** [comment]
This was using only elements already included in the TFT of v4.0.
It's non-functional, but it's available on the TFT already on `dev`. Feel free to play with that and create a PR to merge it back. 😉 
https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/94725493/42b1e180-5a68-4dca-a06d-415a830d2b70

**Comment by edwardtfn** [comment]
#168

**Reply by Floppe** [reply]
Thanks for the font. I did migrate last week, not quite happy with it though. Thinking if I should re-design the entire page.

May I ask how you did the current simulator in the video. Multiple images and timer that changes them?

**Reply by edwardtfn** [reply]
[power.zip](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13216228/power.zip)
Those are status bars with a timer changing it's values.
You can import that page (attached) to your device and see.

**Reply by kroessig** [reply]
Hey,
Can anyone tell me how to import the power page?

**Reply by edwardtfn** [reply]
First of all, be aware this is just a mock-up I've used to validate the User's Interface, so this page is not really functional.

You must use the Nextion editor to import that. There's a box in the top right with all the pages. A right click there will show the option to import a page.
After importing you will have to generate a TFT file and upload that to your device or, as I've done on my video, run the "debug" mode on Nextion Editor and give the commands to go to page "power".

**Comment by Floppe** [comment]
It kind of got more personal than generic now when I switched away from icon font and created a background image. At the moment it looks like this. I'm not so steady on my hands when filming the screen.

![energy page example](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/238729/95ac6842-9c1b-4e0a-a943-bc31291a47ff)

Generator have no energy meter for the moment.
I have no wind production, only thinking if I should get some.
I have geothermal heating so small boxes for how much energy gone into radiators and tap water for the day.
And of course daily grid import/export.

EU screen only.

**Reply by edwardtfn** [reply]
Wow! Amazing!
But makes me a bit nervous thinking about how many diferent sources of energy could be on this screen and the multiple different connections between those... Solar, battery, fuel, wind, grid...

**Reply by Floppe** [reply]
Exactly :)

**Reply by kroessig** [reply]
I'm not really familiar with programming and I only piece together my code half-heartedly. but could you please briefly explain to me how to adapt the blueprint to achieve this? In the Nextion editor I build the layout of the page and assign "variables". Where in the blueprint can I assign values to the variables? Or can I hard program it for testing? 

@Floppe I hope it's not too cheeky, but would you share your page here? I'm really excited!!!!  
The current moving energy flows are only incidental, but the presentation of the values is very felicitous. (btw the unit for daily solar should be kWh).

**Reply by Floppe** [reply]
Well, I have a fork where it is in energy branch. No guarantee it will be for long though as I am not so good with using git and mess it up.

The old screen is also there in energy_old branch.

https://github.com/Floppe/NSPanel_HA_Blueprint

**Reply by edwardtfn** [reply]
> I have a fork where it is in energy branch

Feel free to submit it in a PR. 😉
